dace/transformation/dataflow/redundant_array.py

```diff
@@ -152,8 +147,21 @@ def gnode(nname):
             return graph.nodes()[self.subgraph[nname]]
 
         graph = sdfg.nodes()[self.state_id]
-        in_array = gnode(RedundantArray._in_array)
-        out_array = gnode(RedundantArray._out_array)
+        in_array = self.in_array(sdfg)
+        out_array = self.out_array(sdfg)
+        in_desc = sdfg.arrays[in_array.data]
+        out_desc = sdfg.arrays[out_array.data]
+
+        # If arrays are not of the same shape, modify first array to reshape
+        # instead of removing it
+        if len(in_desc.shape) != len(out_desc.shape) or any(
```

Comment:
Does this solve all issues?

Reply:
It solves redundant arrays that we couldn't remove, for example anything that comes out of numpy with keepdims=True, but I'm not sure how it generalizes to different kinds of slices.